internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -1471,6 +1487,249 @@ func TestMachineSetReconciler_reconcileUnhealthyMachines(t *testing.T) {
+		machineOwnerRemediatedCondition := conditions.Get(m, condition)
+		g.Expect(machineOwnerRemediatedCondition.Status).
+			To(Equal(corev1.ConditionFalse), "%s condition status should be false", condition)
+
+		// Verify the healthy machine continues to not have the MachineOwnerRemediated condition.
+		m = &clusterv1.Machine{}
+		g.Expect(r.Client.Get(ctx, client.ObjectKeyFromObject(healthyMachine), m)).To(Succeed())
+		g.Expect(conditions.Has(m, condition)).
+			To(BeFalse(), "Machine should not have the %s condition set", condition)
+
+		// Test with the current MachineSet.
+		_, err = r.reconcileUnhealthyMachines(ctx, cluster, machineSetCurrent, machines)
+		g.Expect(err).ToNot(HaveOccurred())
+
+		// Verify the unhealthy machine has the updated condition.
+		err = r.Client.Get(ctx, client.ObjectKeyFromObject(unhealthyMachine), m)
+		g.Expect(apierrors.IsNotFound(err)).To(BeTrue())
+	})
+
+	t.Run("should only try to remediate up to MaxInFlight unhealthy", func(t *testing.T) {
+		g := NewWithT(t)
+
+		cluster := &clusterv1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-cluster",
+				Namespace: "default",
+			},
+			Spec: clusterv1.ClusterSpec{},
+		}
+
+		machineDeployment := &clusterv1.MachineDeployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-machinedeployment",
+				Namespace: "default",
+				Annotations: map[string]string{
+					clusterv1.RevisionAnnotation: "10",
+				},
+			},
+			Spec: clusterv1.MachineDeploymentSpec{
+				ClusterName: "test-cluster",
+				Strategy: &clusterv1.MachineDeploymentStrategy{
+					Remediation: &clusterv1.RemediationStrategy{
+						MaxInFlight: ptr.To(intstr.FromInt(3)),
+					},
+				},
+			},
+		}
+
+		machineSet := &clusterv1.MachineSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-machinedeployment-old",
+				Namespace: "default",
+				Labels: map[string]string{
+					clusterv1.MachineDeploymentNameLabel: "test-machinedeployment",
+				},
+				Annotations: map[string]string{
+					clusterv1.RevisionAnnotation: "10",
+				},
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: clusterv1.GroupVersion.String(),
+						Kind:       "MachineDeployment",
+						Name:       "test-machinedeployment",
+					},
+				},
+			},
+			Spec: clusterv1.MachineSetSpec{
+				ClusterName: "test-cluster",
+			},
+		}
+
+		unhealthyMachines := []*clusterv1.Machine{}
+		for i := range 5 {
+			unhealthyMachines = append(unhealthyMachines, &clusterv1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              fmt.Sprintf("unhealthy-machine-%d", i),
+					Namespace:         "default",
+					CreationTimestamp: metav1.Time{Time: metav1.Now().Add(time.Duration(i) * time.Second)},
+				},
+				Status: clusterv1.MachineStatus{
+					Conditions: []clusterv1.Condition{
+						{
+							Type:   clusterv1.MachineOwnerRemediatedCondition,
+							Status: corev1.ConditionFalse,
+						},
+					},
+				},
+			})
+		}
+
```

Comment:
Would be good if we can also extend our coverage to this one
```go
	if maxInFlight <= 0 {
		// No tokens available to remediate machines.
		log.V(3).Info("Remediation strategy is set, and maximum in flight has been reached", "unhealthyMachines", len(machinesToRemediate))
		return ctrl.Result{}, nil
	}
```

(I think this can be done in basically the same way)

Reply:
Expanded the current test with the above

Reply:
Thx, nice!